test/unit/ObjectMakrTests/processFunctions.js

```diff
@@ -0,0 +1,90 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("processes the correct number of functions", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr({
+                inheritance: {
+                    Organism: {
+                        Animal: {
+                            Bird:{
+                                Penguin: {}
+                            },
+                            Mammal: {
+                                Dog: {}
+                            }
+                        },
+                        Plant: {}
+                    }
+                },
+                properties: {
```

Comment:
I'm saying if you explicitly add code in a test, it should be used. You're adding `properties`, `doPropertiesFull`, `indexMap`, and `onMake` here. If they're not being used, don't add them in. It's confusing.